### ISSUE TITLE
Improve error message for `vue/no-dupe-keys`

### ISSUE THREAD

**Issue by vincerubinetti** [vuejs/eslint-plugin-vue#2247]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.45.0
- **eslint-plugin-vue version:** 9.15.1
- **Node version:** 18
- **Operating System:** MacOS M1 

**Please show your full configuration:**

```json5
/* eslint-env node */
require("@rushstack/eslint-patch/modern-module-resolution");

module.exports = {
  root: true,
  extends: [
    "plugin:vue/vue3-recommended",
    "plugin:vuejs-accessibility/recommended",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier/skip-formatting",
  ],
  parserOptions: {
    ecmaVersion: "latest",
  },

  rules: {
    "prettier/prettier": "warn",
  },
};

```

**What did you do?**	

```vue
<script setup lang="ts">
import { debounce } from "lodash"; // <-- error thrown on this line

type Props = {
  /** delay for debounce in ms */
  debounce?: number;
};

const props = defineProps<Props>();

console.log(props.debounce); // i'm using the prop like this, not with any destructuring
</script>

```

**What did you expect to happen?**

These two names are in different contexts, and don't refer to the same thing. I'm not sure if this is expected behavior, but if it is, I find that a little weird, because they don't really conflict.

**Repository to reproduce this issue**
[dupe keys.zip](https://github.com/vuejs/eslint-plugin-vue/files/12152483/dupe.keys.zip)

**Comment by vincerubinetti** [comment]
Actually I think this is intended behavior because there would be a conflict in the `<template>`, even if there isn't a conflict in the `<script>`. Just a side effect of Vue automatically defining props in the `<template>` without any prefix/namespace.

**Comment by vincerubinetti** [comment]
Re-opening this because maybe the error message could be better? Took me a minute to realize what was going on. Or maybe there could be some kind of option to ignore if the duplicates don't actually lead to a conflict? Not sure how hard that would be to implement.

**Comment by FloEdelmann** [comment]
Yes, the rule is working as intended. Could you suggest a better error message?

**Comment by vincerubinetti** [comment]
This might not be consistent with the brevity or granularity of the other error messages, but I might try something like:

> Duplicated key 'debounce' may cause conflict in <script> or <template>.

Alternatively, and perhaps more appropriately, just update the [docs page](https://eslint.vuejs.org/rules/no-dupe-keys.html). Checking this page was the first thing I did, and some clarification could've helped.

> This rule prevents to use duplicated names.

Elaborating on this (in general I think rules descriptions should be more than 1 terse sentence):

> This rule prevents duplicate `props`/`data`/`methods`/etc. key names defined in a component. Even if a key name does not conflict in the `<script>` tag itself, it still may lead to a conflict in the `<template>` because Vue allows directly accessing these keys from there.

**Comment by FloEdelmann** [comment]
I think both changes are a good improvement! Would you fancy opening a PR?
